kbd-mode.el

```diff
@@ -151,6 +151,190 @@ form `@MACRO-NAME' with `kbd-mode-variable-name-face'."
+
+(defun kbd-mode--kill-kmonad-process ()
+  "Kill demo kmonad process.
+
+If there is nothing to kill then do nothing."
+  (when (get-buffer-process "*kmonad*")
+    (kill-process "*kmonad*")))
+
+(defun kbd-mode--old-kmonad-suspend ()
+  "Suspend old kmonad instance."
+  (if kbd-mode-old-kmonad-suspend-command
+      (shell-command  kbd-mode-old-kmonad-suspend-command)
+    (error "Set kbd-mode-old-kmonad-suspend-command variable!")))
+
+(defun kbd-mode--old-kmonad-resume ()
+  "Resume old kmonad instance."
+  (if kbd-mode-old-kmonad-suspend-command
+      (shell-command  kbd-mode-old-kmonad-resume-command)
+    (error "Set kbd-mode-old-kmonad-resume-command variable!")))
+
+;;; Define Demo Minor Mode
+
+;; Keymap
+
+(defvar kbd-mode-demo-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "C-c C-s")  #'kbd-mode-stop-demo)
+    map))
+
+;; Helper Functions
+
+(defun kbd-mode--toggle-demo (window)
+  "Toggle Start/Kill kmonad process.
+
+When exit demo window, kill kmonad process, user will use
+heir normal keyboard layout.
+When re-enter the demo window, kmonad process will started again."
+  (cond ((kbd-mode---demo->other-windowp)
+         (kbd-mode--kill-kmonad-process)
+         (kbd-mode--old-kmonad-resume))
+        ((kbd-mode--valid-configp)
+         (kbd-mode--create-kmonad-process)
+         (kbd-mode--old-kmonad-suspend))
+        (t
+         (kbd-mode--show-error))))
+
+(defun kbd-mode---demo->other-windowp ()
+  "Find which way switch has done.
+From demo window to other window: return t
+From other window to demo window: return nil"
+  (equal (window-buffer (previous-window ))
+         (get-buffer "*kmonad-demo*")))
+
+(defun kbd-mode--show-error ()
+  "Show error with compilation buffer."
+  (quit-window "*kmonad-demo*")
+  (compile (kbd-mode--get-config-validation-command)))
+
+(defun kbd-mode--get-config-validation-command ()
+  "Get validation command for `kbd-mode-demo-file'.
+If FILE is given it'll be checked instead of `kbd-mode-demo-file'."
+  (concat "kmonad -d " kbd-mode-demo-file))
+
+;; Example error ouput
+;; kmonad: Parse error at 36:1:
+;;    |
+;; 36 | jdfkj
+;;    | ^
+;; unexpected 'j'
+;; expecting '(' or end of input
+
+;; Make compilation buffer parseable
+
+(add-to-list 'compilation-error-regexp-alist 'kbd)
+
+(add-to-list 'compilation-error-regexp-alist-alist
+             '(kbd "^kmonad: Parse error at \\([0-9]+\\):\\([0-9]+\\)" nil 1 2))
+
+
+;;;###autoload
+(define-minor-mode kbd-mode-demo-mode
+  "Toggle kmonad demo mode.
+Demo mode has one keybinding to exit the demo mode."
+  :lighter " kbd-demo"
+  :keymap kbd-mode-demo-mode-map
+
+  ;; Initialize variables
+  (when kbd-mode-demo-mode
+    (progn
+      (make-variable-buffer-local 'kbd-mode-demo-file)
+      (unless (kbd-mode--valid-configp)
+        (kbd-mode--show-error))))
+
+  ;; Handle toggle
+  (cond (kbd-mode-demo-mode
+         (add-hook 'window-selection-change-functions #'kbd-mode--toggle-demo nil t)
```

Comment:
are these really necessary?

Reply:
These are necessary for buffer movements.

For example:
User goes to demo to config: Demo's kmonad will be killed, and user's will be restored.
User goes to config to demo: User's kmonad will be killed, and demo's will be created.

But I can remove frame ones, they could be overkill.

Reply:
Seems to me like what I wrote [here](https://github.com/kmonad/kbd-mode/pull/7#pullrequestreview-882650181) becomes more and more relevant